nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java

```diff
@@ -106,15 +106,16 @@
     public static final PropertyDescriptor CLIENT_ID = new PropertyDescriptor.Builder()
         .name("client-id")
         .displayName("Client ID")
-        .required(false)
+        .dependsOn(GRANT_TYPE, CLIENT_CREDENTIALS_GRANT_TYPE)
+        .required(true)
         .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
         .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
         .build();
 
     public static final PropertyDescriptor CLIENT_SECRET = new PropertyDescriptor.Builder()
         .name("client-secret")
         .displayName("Client secret")
-        .dependsOn(CLIENT_ID)
+        .dependsOn(GRANT_TYPE, CLIENT_CREDENTIALS_GRANT_TYPE)
```

Comment:
This is incorrect. **Resource Owner Password Credentials Grant** _also_ requires these parameters.